### ISSUE TITLE
Could not register the following drivers PenImc_cor3.dll running dotnet6 winforms app

### ISSUE THREAD

**Issue by svetovid** [dotnet/windowsdesktop#1485]
* .NET Core Version: 
- SDK 6.0.100-preview.1.21107.2
- Microsoft.NETCore.App 6.0.0-preview.2.21105.12
- Microsoft.WindowsDesktop.App 6.0.0-preview.2.21104.2

* Have you experienced this same bug with .NET Framework?: 
No

**Problem description:**

![image](https://user-images.githubusercontent.com/9212389/107414561-774c3b00-6b12-11eb-90d6-ea52093d7235.png)

Running a simple winforms application on ARM64 causes program crash with message "Could not register the following drivers: PenImc_cor3.dll". After removal of PenImc_cor3.dll from <App>.deps.json the application works again.

As I understand this library belongs to WPF. Why is it in WinForms dependencies?
How to get rid of it during the build/publishing, so we don't need to remove it manually?

**Minimal repro:**
1. dotnet new winforms
2. dotnet publish -c Release -r win-arm64
3. Run on a ARM64 device

**Comment by fabiant3** [comment]
So ARM64 specific. @ryalanms - did we miss adding this DLL?

**Comment by ryalanms** [comment]
I don't believe penimc should ever be loaded by WinForms unless there is WPF hosted content inside a WinForms application.   This looks like a packaging error, but I need to get a repro to confirm.  

PenIMC is a SxS COM dll that is loaded by WPF's PresentationCore when a tablet/touchscreen is detected by WPF.  The embedded manifest in the dll is added to the ActivationContext prior to any calls.  If the binary was missing it should have failed with: "SxS COM registration of PenImc_cor3.dll failed".

**Comment by ryalanms** [comment]
I wasn't able to reproduce the failure with 6.0.100-preview.2.21158.2 on Surface Pro X (a touch device). 

1. dotnet new winforms
2. dotnet publish -c Release -r win-arm64
3. Run on a ARM64 device

@svetovid:  Could you paste the output from "dotnet --info" and share your machine configuration?  Thanks.

**Comment by svetovid** [comment]
Hi @ryalanms,
I specified dotnet info in the description of this issue.
For build I use Windows 10 Pro 1909, x64-based processor Desktop workstation.

After SDK updated to 6.0.100-preview.3.21123.4, the problem seems to be gone.
Thank you.

**Comment by svetovid** [comment]
Hi @ryalanms,

It still doesn't work, but in our big project, which we are migrating to net 6.
It works on x64 and doesn't on arm64 machines.

Why **PenImc_cor3.dll** is present in the publish folder even if it's a WinForms application?

**Comment by svetovid** [comment]
Hi,

In our solution we have a step which uses Regsvr32.exe to register native dlls. It works fine on x64 machines but fails with mentioned error on ARM64.

We removed it from our pipeline, so this issue can be closed.

Thank you.

**Comment by danielkornev** [comment]
same issue with WPF 8.0 app running on ARM64 machine (SP11)
